Dev endpoint bugs

Looby72 opened the pull request:
# Made some major bug fixes in the API
## Send new session Cookie on every successful request #97 
- a request is successful (codes 200 or 201) then the JWT is sent back to the client with a new cookie, which lasts one hour

## Bug fixes in solution endpoint
- solution_user attribute is now calculated from the backend from the JWT #100
- some other bug fixes, and changing some examples in the documentation

## fixed changing roles in user endpoint
- demoting admin account to a user account is working now #99
- user_role is also changed in the database now (previously that was not working)

## removing some test cases
- removed all test cases trying to create an admin account (I thought I have done that already)

mlinke-ai commented:
Would it be a bad idea if we returned a new cookie also on unsuccessful requests?

Looby72 commented:
We could do that. Then we have to discuss what happens when the client sends no (or no valid) JWT.

- If no JWT is sent I just would not attach a new cookie.
- If the cookie contains a JWT I would just send it back. (no matter if its valid or  not). I could also check if the token contains real user data or not but then I have to decode the JWT again. Maybe I could also implement that into the authorize function somehow.

Looby72 commented:
# Made some important changes
## authorize function now returns user_id
- we don't have to decode the JWT more than once

## returning user_info in a different manner
- if the client requests its own user date by sending an empty `GET` to `/user` the user data will be returned as single JSON dict (not mapped to the user_id), since this is unnecessary for one retuned element

## renewing cookies for every HTTP response
- wrote a new function `makeResponseNewCookie` that returns a response object with renewed cookie `Expires` value
- only returns a renewed cookie a value for key `token` exists (only one value)
- so if a wrong JWT is sent by the client it will also be renewed

## changed login process as discussed
- changed required argument from user_name to user_mail
- that means, the e-Mail address is now needed to log into an account
- also changed all test cases that somehow needs to log into an account
- changed documentation according to that
